File: app/src/main/java/fr/free/nrw/commons/wikidata/model/edit/Edit.java
package fr.free.nrw.commons.wikidata.model.edit;

import androidx.annotation.Nullable;
import fr.free.nrw.commons.wikidata.mwapi.MwPostResponse;

public class Edit extends MwPostResponse {
    @Nullable private Result edit;

    @Nullable public Result edit() {
    return edit;
    }

    public class Result {
        @Nullable private String result;
        @Nullable private String code;
        @Nullable private String info;
        @Nullable private String warning;

        public boolean editSucceeded() {
            return "Success".equals(result);
        }

        @Nullable public String code() {
            return code;
        }

        @Nullable public String info() {
            return info;
        }

        @Nullable public String warning() {
            return warning;
        }

    }
}

File: app/src/main/java/fr/free/nrw/commons/wikidata/model/edit/Edit.kt
package fr.free.nrw.commons.wikidata.model.edit

import fr.free.nrw.commons.wikidata.mwapi.MwPostResponse

class Edit : MwPostResponse() {
    private val edit: Result? = null

    fun edit(): Result? = edit

    class Result {
        private val result: String? = null
        private val code: String? = null
        private val info: String? = null
        private val warning: String? = null

        fun editSucceeded(): Boolean =
            "Success" == result

        fun code(): String? = code

        fun info(): String? = info

        fun warning(): String? = warning
    }
}
